cmd/theatre-envconsul/main.go

```diff
@@ -193,6 +219,25 @@ func mainError(ctx context.Context, command string) (err error) {
 			return errors.Wrap(err, "failed to decode envconsul environment variables")
 		}
 
+		// For every file reference, write the value resolved by envconsul to the path in
+		// filePaths
+		for key, value := range filePaths {
+			if value == pathTBD {
+				// generate file path prefixed by key
+				tempFilePath, err := ioutil.TempFile("", fmt.Sprintf("%s-*", key))
+				if err != nil {
+					return errors.Wrap(err, fmt.Sprintf("failed to write temporary file for key %s", key))
+				}
+				value = tempFilePath.Name()
+			}
+
+			// write file with value of envMap[key]
+			if err := ioutil.WriteFile(value, []byte(env[key]), 0644); err != nil {
```

Comment:
this does not work if value (path) is nested e.g /tmp/secrets/secretfile.

Reply:
WDYT about using `os.MkdirAll` before this, to ensure that the path structure is available?